### ISSUE TITLE
Version 16.11.0 Preview 2.0 - InvalidOperationException Unexpected false

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#54713]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Version-16110-Preview-20---InvalidOpe/1470050)._

---
Highlighting the "1" in the following code throws a yellow bar error with a type of InvalidOperation 'Unexpected false'.

```
namespace Namespace
{
    public static class Class
    {
        private async FOO BAR
        {
            using (new ABC())
            {
            List<string> ParametersValues = new List<string>
                {
                "1"
                }
            }
        }
    }
}
```

```
System.InvalidOperationException : Unexpected false
   at Roslyn.Utilities.Contract.Fail(String message)
   at Roslyn.Utilities.Contract.ThrowIfFalse(Boolean condition)
   at Microsoft.CodeAnalysis.CSharp.ExtractMethod.CSharpMethodExtractor.CSharpCodeGenerator.ExpressionCodeGenerator.GetCallSiteContainerFromExpression()
   at Microsoft.CodeAnalysis.CSharp.ExtractMethod.CSharpMethodExtractor.CSharpCodeGenerator.ExpressionCodeGenerator.GetOutermostCallSiteContainerToProcess(CancellationToken cancellationToken)
   at async Microsoft.CodeAnalysis.ExtractMethod.MethodExtractor.CodeGenerator`3.GenerateAsync[TStatement,TExpression,TNodeUnderContainer](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ExtractMethod.MethodExtractor.ExtractMethodAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ExtractMethod.AbstractExtractMethodService`3.ExtractMethodAsync[TValidator,TExtractor,TResult](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.ExtractMethod.ExtractMethodCodeRefactoringProvider.ExtractMethodAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.ExtractMethod.ExtractMethodCodeRefactoringProvider.GetCodeActionsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.ExtractMethod.ExtractMethodCodeRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
```

---
#### Feedback Bot on 7/7/2021, 11:49 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by jinujoseph** [comment]
@akhera99 can you validate if this is fixed , also https://github.com/dotnet/roslyn/issues/35435 seems in the same codepath

**Comment by akhera99** [comment]
Fixed with #53031
